.github/workflows/test.yml

```diff
@@ -42,3 +40,48 @@ jobs:
 
     - name: run unit tests (includes lint)
       run: yarn test
+
+
+  node-integration-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ['18.x', '20.x']
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
```

Comment:
Since we're using `setup-node@v3` and yarn <= 2, I think we can use caching to speed up dependency installs
```yaml
   with:
        node-version: ${{ matrix.node-version }}
        cache: 'yarn'
```
https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

Reply:
I couldn't find it, but does it say if it will upgrade to the latest version within semver, or if it finds a version in the cache that matches semver it will just use it, even if it's not the latest? I skimmed and I found a flag for managing the latest version of Node but it doesn't seem to apply to the packages. I don't know if matters that much though, just something to remember if there's any discrepancy between CI and local test runs.

Reply:
@dlarocque Done!